File: extract_pptx.js
const fs = require('fs');
const path = require('path');

const extractedDir = path.join(__dirname, 'temp_pptx', 'extracted');
const slidesDir = path.join(extractedDir, 'ppt', 'slides');

// 获取所有幻灯片文件
const slideFiles = fs.readdirSync(slidesDir)
  .filter(f => f.startsWith('slide') && f.endsWith('.xml'))
  .sort((a, b) => {
    const aNum = parseInt(a.replace('slide', '').replace('.xml', ''));
    const bNum = parseInt(b.replace('slide', '').replace('.xml', ''));
    return aNum - bNum;
  });

console.log(`找到 ${slideFiles.length} 张幻灯片`);
console.log('='.repeat(50));

slideFiles.forEach((filename, index) => {
  const slideNum = filename.replace('slide', '').replace('.xml', '');
  const filePath = path.join(slidesDir, filename);
  
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    
    // 简单正则提取 <a:t> 标签内的文本
    // 注意：这个正则不处理嵌套标签或属性中的 <a:t>，但对大多数PPTX足够
    const textMatches = content.match(/<a:t[^>]*>([^<]+)<\/a:t>/g);
    
    let texts = [];
    if (textMatches) {
      texts = textMatches.map(tag => {
        // 提取标签内的文本
        const match = tag.match(/<a:t[^>]*>([^<]+)<\/a:t>/);
        return match ? match[1].trim() : '';
      }).filter(text => text.length > 0);
    }
    
    console.log(`幻灯片 ${slideNum}:`);
    if (texts.length > 0) {
      texts.forEach((text, i) => {
        console.log(`  ${i + 1}. ${text}`);
      });
    } else {
      console.log(`  (无文本内容)`);
    }
    console.log();
  } catch (error) {
    console.log(`幻灯片 ${slideNum} 读取错误: ${error.message}`);
    console.log();
  }
});

File: extract_pptx.py
import os
import xml.etree.ElementTree as ET
from pathlib import Path

# 路径
extracted_dir = r"C:\Users\Komorebi\.openclaw\workspace\temp_pptx\extracted"
slides_dir = os.path.join(extracted_dir, "ppt", "slides")

# 命名空间
namespaces = {
    'a': 'http://schemas.openxmlformats.org/drawingml/2006/main',
    'p': 'http://schemas.openxmlformats.org/presentationml/2006/main'
}

# 注册命名空间以便查找
for prefix, uri in namespaces.items():
    ET.register_namespace(prefix, uri)

# 遍历所有幻灯片文件
slide_files = sorted(Path(slides_dir).glob('slide*.xml'))
print(f"找到 {len(slide_files)} 张幻灯片")
print("=" * 50)

for slide_file in slide_files:
    slide_num = slide_file.stem.replace('slide', '')
    try:
        tree = ET.parse(slide_file)
        root = tree.getroot()
        
        # 查找所有文本节点
        text_elements = root.findall('.//a:t', namespaces)
        texts = [elem.text.strip() for elem in text_elements if elem.text and elem.text.strip()]
        
        print(f"幻灯片 {slide_num}:")
        if texts:
            for i, text in enumerate(texts, 1):
                print(f"  {i}. {text}")
        else:
            print("  (无文本内容)")
        print()
    except Exception as e:
        print(f"幻灯片 {slide_num} 解析错误: {e}")
        print()